From 55f8d0de1d8d807069f140ec6999afcab2338041 Mon Sep 17 00:00:00 2001
From: Kevin Chen <kevin_chen@aspeedtech.com>
Date: Thu, 21 Aug 2025 17:21:20 +0800
Subject: [PATCH 11/32] riscv: dts: Add EDAF bridge global cfg node

Add new EDAF bridge configuration nodes to support additional
device-tree properties and enhance functionality.

Introduce references to the EDAF bridge configuration in the
device-tree files for better modularity and maintainability.

Change-Id: Ibe75a30222c7305148728f93306e7a08b31151c9
Signed-off-by: Kevin Chen <kevin_chen@aspeedtech.com>
(cherry picked from commit e0c2b2425bf287ce830c265b7a17d8fceeac131f)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 arch/riscv/dts/ast2700-ibex.dts    | 1 +
 arch/riscv/dts/ast2700-u-boot.dtsi | 4 ++++
 arch/riscv/dts/ast2700.dtsi        | 7 +++++++
 3 files changed, 12 insertions(+)

diff --git a/arch/riscv/dts/ast2700-ibex.dts b/arch/riscv/dts/ast2700-ibex.dts
index da213ab0ef..7cd3520723 100644
--- a/arch/riscv/dts/ast2700-ibex.dts
+++ b/arch/riscv/dts/ast2700-ibex.dts
@@ -241,6 +241,7 @@
 	espi-base = <0x0 0x14c05000>;
 #if 0 //DDR mode using edaf0 reserved memory
 	edaf-ddr-mode;
+	edaf-earse-with-1;
 	mem-base = <&edaf0>;
 #else //Flash mode using SPI0
 	mem-base = <0x1 0x80000000>; //SPI0
diff --git a/arch/riscv/dts/ast2700-u-boot.dtsi b/arch/riscv/dts/ast2700-u-boot.dtsi
index b053113be9..2ece13966c 100644
--- a/arch/riscv/dts/ast2700-u-boot.dtsi
+++ b/arch/riscv/dts/ast2700-u-boot.dtsi
@@ -118,6 +118,10 @@
 			};
 		};
 
+		edaf_bridge_cfg: edaf-bridge-cfg@140ec100 {
+			bootph-all;
+		};
+
 		sli1: sli@14c1e000 {
 			bootph-all;
 		};
diff --git a/arch/riscv/dts/ast2700.dtsi b/arch/riscv/dts/ast2700.dtsi
index 474e9f25c7..437ff14661 100644
--- a/arch/riscv/dts/ast2700.dtsi
+++ b/arch/riscv/dts/ast2700.dtsi
@@ -243,6 +243,11 @@
 			};
 		};
 
+		edaf_bridge_cfg: edaf-bridge-cfg@140ec100 {
+			compatible = "aspeed,ast2700-edaf-bridge-cfg";
+			reg = <0x140ec100 0x100>;
+		};
+
 		sli1: sli@14c1e000 {
 			compatible = "aspeed,ast2700-sli1";
 			reg = <0x14c1e000 0x500>;
@@ -278,6 +283,7 @@
 			compatible = "aspeed,ast2700-edaf-bridge0";
 			reg = <0x14c38000 0x200>;
 			aspeed,scu1 = <&syscon1>;
+			aspeed,edaf_bridge_cfg = <&edaf_bridge_cfg>;
 			status = "disabled";
 		};
 
@@ -285,6 +291,7 @@
 			compatible = "aspeed,ast2700-edaf-bridge1";
 			reg = <0x14c38200 0x200>;
 			aspeed,scu1 = <&syscon1>;
+			aspeed,edaf_bridge_cfg = <&edaf_bridge_cfg>;
 			status = "disabled";
 		};
 
-- 
2.34.1

